--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -3443,10 +3443,16 @@ static void vmap_debug_free_range(unsigned long start, unsigned long end)
 #ifdef CONFIG_KPM
 static void free_unmap_vmap_area(struct vmap_area *va)
 {
-	if (v->flags & VM_LOWMEM)
+	/* 使用 VM_MAP 替代 VM_LOWMEM */
+#ifdef VM_LOWMEM
+	if (v->flags & VM_LOWMEM) 
 		kpm_free_page(addr, size);
-	else
+	else 
+		vunmap_page_range(addr, addr + size);
+#else
+	if (v->flags & VM_MAP)
+		kpm_free_page(addr, size);
+	else
 		vunmap_page_range(addr, addr + size);
+#endif
 }
 #endif
