diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 75081da0bf19..8e9b7d8b9a19 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -3451,9 +3451,6 @@ static int s_show(struct seq_file *m, void *p)
 
 	v = va->vm;
 
-	if (v->flags & VM_LOWMEM)
-		return 0;
-
 	seq_printf(m, "0x%pK-0x%pK %7ld",
 		v->addr, v->addr + v->size, v->size);
 
diff --git a/net/wireguard/ratelimiter.c b/net/wireguard/ratelimiter.c
index [原索引]..[新索引] 100644
--- a/net/wireguard/ratelimiter.c
+++ b/net/wireguard/ratelimiter.c
@@ -18,6 +18,7 @@
 #include "ratelimiter.h"
 #include <linux/siphash.h>
 #include <linux/mm.h>
+#include <linux/sysinfo.h>
 #include <linux/slab.h>
 #include <net/ip.h>
 
@@ -174,6 +175,7 @@ void wg_ratelimiter_uninit(void)
 
 int wg_ratelimiter_init(void)
 {
+	struct sysinfo i;
 	mutex_lock(&init_lock);
 	if (++init_refcnt != 1)
 		goto out;
@@ -185,10 +187,10 @@ int wg_ratelimiter_init(void)
 	 * we borrow their wisdom about good table sizes on different systems
 	 * dependent on RAM. This calculation here comes from there.
 	 */
-	table_size = (totalram_pages() > (1U << 30) / PAGE_SIZE) ? 8192 :
-		max_t(unsigned long, 16, roundup_pow_of_two(
-			(totalram_pages() << PAGE_SHIFT) /
-			(1U << 14) / sizeof(struct hlist_head)));
+	si_meminfo(&i);
+	table_size = (i.totalram > (1U << 30) / PAGE_SIZE) ? 8192 :
+		     max_t(unsigned long, 16, roundup_pow_of_two(
+			     (i.totalram << PAGE_SHIFT) / (1U << 14) / sizeof(struct hlist_head)));
 	max_entries = table_size * 8;
 
 	table_v4 = kvcalloc(table_size, sizeof(*table_v4), GFP_KERNEL);

diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 7aad09b65cad..40292b158fe2 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -578,35 +578,16 @@ static void __free_vmap_area(struct vmap_area *va)
 {
 	BUG_ON(RB_EMPTY_NODE(&va->rb_node));
 
-	if (free_vmap_cache) {
-		if (va->va_end < cached_vstart) {
-			free_vmap_cache = NULL;
-		} else {
-			struct vmap_area *cache;
-			cache = rb_entry(free_vmap_cache, struct vmap_area, rb_node);
-			if (va->va_start <= cache->va_start) {
-				free_vmap_cache = rb_prev(&va->rb_node);
-				/*
-				 * We don't try to update cached_hole_size or
-				 * cached_align, but it won't go very wrong.
-				 */
-			}
-		}
-	}
-	rb_erase(&va->rb_node, &vmap_area_root);
-	RB_CLEAR_NODE(&va->rb_node);
-	list_del_rcu(&va->list);
-
 	/*
-	 * Track the highest possible candidate for pcpu area
-	 * allocation.  Areas outside of vmalloc area can be returned
-	 * here too, consider only end addresses which fall inside
-	 * vmalloc area proper.
+	 * Remove from the busy tree/list.
 	 */
-	if (va->va_end > VMALLOC_START && va->va_end <= VMALLOC_END)
-		vmap_area_pcpu_hole = max(vmap_area_pcpu_hole, va->va_end);
+	unlink_va(va, &vmap_area_root);
 
-	kfree_rcu(va, rcu_head);
+	/*
+	 * Merge VA with its neighbors, otherwise just add it.
+	 */
+	merge_or_add_vmap_area(va,
+		&free_vmap_area_root, &free_vmap_area_list);
 }
 
 /*
