diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 75081da0bf19..8e9b7d8b9a19 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -3451,9 +3451,6 @@ static int s_show(struct seq_file *m, void *p)
 
 	v = va->vm;
 
-	if (v->flags & VM_LOWMEM)
-		return 0;
-
 	seq_printf(m, "0x%pK-0x%pK %7ld",
 		v->addr, v->addr + v->size, v->size);
 
--- a/net/wireguard/ratelimiter.c
+++ b/net/wireguard/ratelimiter.c
@@ -184,6 +184,7 @@ int wg_ratelimiter_init(void)
 	if (++init_refcnt != 1)
 		goto out;
 
+	unsigned long totalram = totalram_pages();
 	entry_cache = KMEM_CACHE(ratelimiter_entry, 0);
 	if (!entry_cache)
 		goto err;
@@ -193,9 +194,9 @@ int wg_ratelimiter_init(void)
 	 * we borrow their wisdom about good table sizes on different systems
 	 * dependent on RAM. This calculation here comes from there.
 	 */
-	table_size = (totalram_pages() > (1U << 30) / PAGE_SIZE) ? 8192 :
+	table_size = (totalram > (1U << 30) / PAGE_SIZE) ? 8192 :
 		max_t(unsigned long, 16, roundup_pow_of_two(
-			(totalram_pages() << PAGE_SHIFT) /
+			(totalram << PAGE_SHIFT) /
 			(1U << 14) / sizeof(struct hlist_head)));
 	max_entries = table_size * 8;
