--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -3451,9 +3451,6 @@ static int s_show(struct seq_file *m, void *p)
 
 	v = va->vm;
 
-	if (v->flags & VM_LOWMEM)
-		return 0;
-
 	seq_printf(m, "0x%pK-0x%pK %7ld",
 		v->addr, v->addr + v->size, v->size);

--- a/mm/swap.c	2025-04-07 20:30:43.154752552 +0800
+++ b/mm/swap.c	2025-04-07 20:27:34.065681866 +0800
@@ -981,7 +981,7 @@
  */
 void __init swap_setup(void)
 {
-	unsigned long megs = totalram_pages >> (20 - PAGE_SHIFT);
+	unsigned long megs = totalram_pages() >> (20 - PAGE_SHIFT);
 #ifdef CONFIG_SWAP
 	int i;

--- a/drivers/net/wireguard/ratelimiter.c
+++ b/drivers/net/wireguard/ratelimiter.c
@@ -18,6 +18,7 @@
 #include "ratelimiter.h"
 #include <linux/siphash.h>
 #include <linux/mm.h>
+#include <linux/sysinfo.h>
 #include <linux/slab.h>
 #include <net/ip.h>
 
@@ -174,6 +175,7 @@ void wg_ratelimiter_uninit(void)
 
 int wg_ratelimiter_init(void)
 {
+	struct sysinfo i;
 	mutex_lock(&init_lock);
 	if (++init_refcnt != 1)
 		goto out;
@@ -185,10 +187,10 @@ int wg_ratelimiter_init(void)
 	 * we borrow their wisdom about good table sizes on different systems
 	 * dependent on RAM. This calculation here comes from there.
 	 */
-	table_size = (totalram_pages() > (1U << 30) / PAGE_SIZE) ? 8192 :
-		max_t(unsigned long, 16, roundup_pow_of_two(
-			(totalram_pages() << PAGE_SHIFT) /
-			(1U << 14) / sizeof(struct hlist_head)));
+	si_meminfo(&i);
+	table_size = (i.totalram > (1U << 30) / PAGE_SIZE) ? 8192 :
+		     max_t(unsigned long, 16, roundup_pow_of_two(
+			     (i.totalram << PAGE_SHIFT) / (1U << 14) / sizeof(struct hlist_head)));
 	max_entries = table_size * 8;
 
 	table_v4 = kvcalloc(table_size, sizeof(*table_v4), GFP_KERNEL);
